### DISCUSSION TITLE
OpenAPI transpilation

### DISCUSSION THREAD

**Discussion by dils2k** [jsightapi/specification#9] [Ideas]
Would it be a good idea to create a tool, that can transpile JSight to OpenAPI syntax? This way JSight would have compatability with all existing tools, that work with OpenAPI.

Right now JSight lacks codegen for me. If I could generate OpenAPI spec, then I could use the generated spec to generate both clients and servers. I think that would be pretty cool.

**Comment by andrew-platonov** [comment]
Hi! You are absolutely right.

We are currently working on such a converter. 

You can see the progress in these repos: 
- https://github.com/jsightapi/jsight-api-core, 
- https://github.com/jsightapi/jsight-schema-core, 
- https://github.com/jsightapi/online-editor-frontend.

We plan to release the OAS converter in April 2024.
